### PR TITLE
Revise setup and installation guide

### PR DESCRIPTION
**Pull request by dcfidalgo**
A quick revision of our Setup&Installation guide in the docs. If you are ok with the changes, i would copy some of them over to the repo's readme:
- switched order, first install client, than launch webapp
- moved the descriptions to the sections rather than putting them in the enumerations
- changed some wordings

### CODE REVIEW

File: docs/getting_started/setup&installation.rst [hunk 1]
```diff
@@ -5,60 +5,78 @@ Setup and installation
 
 In this guide, we will help you to get up and running with Rubrix. Basically, you need to:
 
+1. Install the Python client
+2. Launch the webapp
```
Comment (ignacioct) [inline docs/getting_started/setup&installation.rst]:
```suggestion
2. Launch the web app
```

Separated sounds better to me

File: docs/getting_started/setup&installation.rst [hunk 2]
```diff
@@ -5,60 +5,78 @@ Setup and installation
...
 
-#. Setup and launch the webapp, whose main external dependency is `Elasticsearch <https://www.elastic.co/guide/en/elasticsearch/reference/current/install-elasticsearch.html#install-elasticsearch>`_\ , Rubrix's main persistence layer.
-#. Install the Python library.
+1. Install the Rubrix python client
```
Comment (ignacioct) [inline docs/getting_started/setup&installation.rst]:
```suggestion
1. Install the Rubrix Python client
```

File: docs/getting_started/setup&installation.rst [hunk 3]
```diff
@@ -5,60 +5,78 @@ Setup and installation
...
+This is the recommended way because it automatically includes an
+`Elasticsearch <https://www.elastic.co/elasticsearch/>`_ instance, Rubrix's main persistent layer.
 
+Executing the server code manually
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-#. If you do not have an existing Elasticsearch installation,  install `Elasticsearch <https://www.elastic.co/guide/en/elasticsearch/reference/7.12/install-elasticsearch.html>`_ and launch your Elasticsearch instance.
-#. Install the Rubrix Python library following the `guide <>`_ below.
-#. Launch a local instance of the Rubrix webapp ``python -m rubrix.server``. By default, the Rubrix server will look for your ES endpoint URL at ``http://localhost:9200`` if you want to customize this, you can setup the ``ELASTICSEARCH`` env variable with your ES endpoint URL.
+When executing the server coder manually you need to provide an
```
Comment (ignacioct) [inline docs/getting_started/setup&installation.rst]:
```suggestion
When executing the server code manually you need to provide an
```

Isn't it code?

File: docs/getting_started/setup&installation.rst [hunk 4]
```diff
@@ -5,60 +5,78 @@ Setup and installation
...
 
-You should be able to access Rubrix via `http://localhost:6900/ <http://localhost:6900/>`_\ , and you can also check the API docs at `http://localhost:6900/api/docs <http://localhost:6900/api/docs>`_.
+.. code-block:: bash
 
-2. Install the Rubrix Python library
-------------------------------------
+   pip install rubrix[server]
 
-First, make sure you have Python 3.6-3.8 and ``pip`` installed. 
+3. Launch a local instance of the Rubrix webapp
```
Comment (ignacioct) [inline docs/getting_started/setup&installation.rst]:
```suggestion
3. Launch a local instance of the Rubrix web app
```

Same as above

**Review by ignacioct — COMMENTED** [review]
Very minor comments

**Comment by frascuchon** [comment]
Like @dcfidalgo mentions in [this comment](https://github.com/recognai/rubrix/pull/90#issuecomment-856843679), an  `Install from sources` sections could be defined, including provided conda environment_dev and build ui statics

**Comment by ignacioct** [comment]
> Like @dcfidalgo mentions in [this comment](https://github.com/recognai/rubrix/pull/90#issuecomment-856843679), an `Install from sources` sections could be defined, including provided conda environment_dev and build ui statics

Are we using the Docker compose up startup as the recommended one, or the python -m ... ?

**Comment by dvsrepo** [comment]
Yes, that's a good point, we can add dev installation instructions, in that
case and elastic local instance is needed.

For other users I'd still recommended docker-compose, but we can definitely
improve the guide.

Also, Paco mentioned that Elasticsearch can be installed with brew on Mac
and there's an installer for Windows. If possible I'd like to see that
reflected in the updated setup guide.

Finally, see for the library install with pip, please see if it's enough
with what we have or we should be more precise, recommend conda envs etc.

**Comment by frascuchon** [comment]
> Like @dcfidalgo mentions in [this comment](https://github.com/recognai/rubrix/pull/90#issuecomment-856843679), an `Install from sources` sections could be defined, including provided conda environment_dev and build ui statics

mlflow guys have a similar problem https://github.com/mlflow/mlflow/issues/3631

**Comment by dcfidalgo** [comment]
I added a new section to our `Community` chapter in the docs: **Developer documentation**.
The idea is that in this section we provide guides for the development of Rubrix. For now i included the development setup, and how to build the documentation. 

> Also, Paco mentioned that Elasticsearch can be installed with brew on Mac
and there's an installer for Windows. If possible I'd like to see that
reflected in the updated setup guide.

Yeah, one thing is to install ES, another one is to set it up ... i don't think we can cover the latter one in this guide, and we already provide a link to the corresponding parts of the ES documentation.

> Finally, see for the library install with pip, please see if it's enough
with what we have or we should be more precise, recommend conda envs etc.

Since the Rubrix client is quite light weight on dependencies, i don't think we have to recommend using a dedicated environment, to me it just feels like an obstacle to get started quickly.

**Comment by dvsrepo** [comment]
Perfect!

**Comment by frascuchon** [comment]
> Yeah, one thing is to install ES, another one is to set it up ... i don't think we can cover the latter one in this guide, and we already provide a link to the corresponding parts of the ES documentation.

The windows installer package make a guided installation with the famous Windows "Next" approach. At the end of the process you have a fully up and running elasticsearch service without effort.

For mac users, the homebrew command magic make installation and configuration easy too. After install, just run `elasticesarch`  command and everything work like a charm.

For linux users,....You've asked for it. 

In general, for users without a minimal docker background, I think is easier install elasticsearch instance locally and then launch rubrix server, but it's my humble opinion

**Comment by dcfidalgo** [comment]
Ok, i added the links to the homebrew formulae/msi package installation guides. I never tried it myself, just quickly saw those guides and had the feeling it was not so straight forward, but apparently the first impression was wrong.

File: docs/getting_started/setup&installation.rst [hunk 5]
```diff
@@ -5,60 +5,81 @@ Setup and installation
...
-You can launch your docker-contained environment with the following command:
+Using ``docker-compose`` (recommended)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For this method you first need to install `Docker Compose <https://docs.docker.com/compose/install/>`_.
+Then you can launch the docker-contained webapp with the following command:
 
 .. code-block:: bash
 
    wget -O docker-compose.yml https://raw.githubusercontent.com/recognai/rubrix/master/docker-compose.yaml && docker-compose up
```
Comment (frascuchon) [inline docs/getting_started/setup&installation.rst:38]: The docker-compose installation should be a bit more detailed. Users should create a folder and download compose file there, for eventuals problems or app down actions. 

More like elasticsear open distro does: https://opendistro.github.io/for-elasticsearch/downloads.html#try

**Review by dvsrepo — APPROVED** [review]
💯

**Pull request merged**